src/deepsparse/analyze.py

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import logging
+from typing import Any, Dict, List, Optional
+
+import click
+import onnx
+from onnx import ModelProto
+
+import pandas as pd
+from deepsparse import model_debug_analysis
+from deepsparse.benchmark.benchmark_model import benchmark_model
+from deepsparse.utils import generate_random_inputs, model_to_path
+from sparsezoo.analyze import (
+    BenchmarkResult,
+    BenchmarkScenario,
+    ImposedSparsificationInfo,
+    ModelAnalysis,
+    NodeInferenceResult,
+)
+from sparsezoo.analyze.cli import analyze_options, analyze_performance_options
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+@click.command()
+@analyze_options
+@analyze_performance_options
+def main(
+    model_path: str,
+    save: str,
+    batch_size_throughput: int,
+    benchmark_engine: str,
+    **kwargs,
+):
+    """
+    DeepSparse Performance analysis for ONNX models.
+
+    MODEL_PATH: can be a SparseZoo stub, or local path to a
+    deployment-directory or ONNX model
+
+    Examples:
+
+    - Run model analysis on resnet
+
+        deepsparse.analyze ~/models/resnet50.onnx
+    """
+    logging.basicConfig(level=logging.INFO)
+
+    for unimplemented_feat in (
```

Comment:
what's the goal of this logic?

Reply:
It's a placeholder, to show a better error message for unimplemented features, as they'll be a part of the CLI, the CLI is shared accross all analyze api's, so it's on the repo level to describe features that are not (yet) implemented; in the future we'll have all these features implemented